regression/goto-instrument/restrict-function-pointer-to-complex-expression/test.c

```diff
@@ -0,0 +1,59 @@
+#include <assert.h>
+
+typedef int (*fptr_t)(int);
+
+fptr_t get_f(void);
+fptr_t get_g(void);
+
+void use_fg(int choice, fptr_t fptr, fptr_t gptr)
+{
+  assert((choice ? fptr : gptr)(10) == 10 + choice);
+}
+
+// this is just here to distinguish the behavior from FP removal, which'd include g
+int g_always_false_cond = 0;
+
+int main(void)
+{
+  use_fg(0, get_f(), get_g());
```

Comment:
I'm a bit wary that constant propogation will make this example work for free? Consider using a nondet int for the `choice` param

Reply:
The FP removal here happens before symex, so if constant propagation works for this we did our job right.